.ci/snapshot.sh

```diff
@@ -23,12 +23,12 @@ echo $PATH
 java -version
 
 publishArg=''
-if [[ "$dry_run" == "true" ]] ; then
-    echo "--- Build and publish the snapshot :package: (dry-run)"
-    publishArg='publishAllPublicationsToDryRunRepository'
-else
+if [[ "$dry_run" == "false" ]] ; then
```

Comment:
I prefer to play safe and do if `dry_run=false` is explicitly set. Otherwise, it's a dry-run mode.

Reply:
Here do you mean it should default to dry-run and only use the dry-run when explicitly executed with `dry_run=false` ? Maybe updating the default value in the buildkite input pipeline would be relevant for consistency.

Reply:
I meant that the release/snapshot should only run if `dry_run=false`.

> Maybe updating the default value in the buildkite input pipeline would be relevant for consistency.

That's already in the GitHub action
- https://github.com/elastic/elastic-otel-java/blob/091edb229a7b2c6baf90331c9a605265dbeb98d8/.github/workflows/release.yml#L86


Buildkite pipelines are not aimed to be ran manually, it's possible to use them, but I only did it to validate these changes. 


There is no concept of buildkit input values but build environment variables. Do you still want to default to `false`? It's possible to do it, but I dson't see the value now with my current proposal

Reply:
Thanks for the clarification here, making the dry-run the default behavior here only applies to the buildkite pipeline when the env variable `dry_run` is not provided or has a value that is not `true` or `false`, and there is no need to also do the same at the GH actions workflow.

When the BK job is triggered from GH actions, the parameter is always provided with either `true` or `false` value.